src/schemathesis/specs/openapi/converter.py

```diff
@@ -108,9 +108,23 @@ def is_read_only(schema: dict[str, Any] | bool) -> bool:
     return schema.get("readOnly", False)
 
 
+@overload
 def to_json_schema_recursive(
     schema: dict[str, Any], nullable_name: str, is_response_schema: bool = False, update_quantifiers: bool = True
-) -> dict[str, Any]:
+) -> dict[str, Any]: ...  # pragma: no cover
+
+
+@overload
+def to_json_schema_recursive(
+    schema: bool, nullable_name: str, is_response_schema: bool = False, update_quantifiers: bool = True
+) -> bool: ...  # pragma: no cover
+
+
+def to_json_schema_recursive(
```

Comment:
Move to `core.jsonschema` & refactor? maybe move `update_quantifier` to `core` too?